tools/tests/py2fgen/test_diffusion_wrapper.py

```diff
@@ -6,146 +6,174 @@
-
-    # input data - gt4py fields
-    theta_ref_mc = as_field((dims.CellDim, dims.KDim), theta_ref_mc)
-    wgtfac_c = as_field((dims.CellDim, dims.KDim), wgtfac_c)
-    vct_a = as_field((dims.KDim,), vct_a)
-    e_bln_c_s = as_field((dims.CellDim, dims.C2EDim), e_bln_c_s)
-    geofac_div = as_field((dims.CellDim, dims.C2EDim), geofac_div)
-    geofac_grg_x = as_field((dims.CellDim, dims.C2E2CODim), geofac_grg_x)
-    geofac_grg_y = as_field((dims.CellDim, dims.C2E2CODim), geofac_grg_y)
-    geofac_n2s = as_field((dims.CellDim, dims.C2E2CODim), geofac_n2s)
-    nudgecoeff_e = as_field((dims.EdgeDim,), nudgecoeff_e)
-    rbf_coeff_1 = as_field((dims.VertexDim, dims.V2EDim), rbf_coeff_1)
-    rbf_coeff_2 = as_field((dims.VertexDim, dims.V2EDim), rbf_coeff_2)
-    dwdx = as_field((dims.CellDim, dims.KDim), dwdx)
-    dwdy = as_field((dims.CellDim, dims.KDim), dwdy)
-    hdef_ic = as_field((dims.CellDim, dims.KDim), hdef_ic)
-    div_ic = as_field((dims.CellDim, dims.KDim), div_ic)
-    mask_hdiff = as_field((dims.CellDim, dims.KDim), mask_hdiff)
-    zd_diffcoef = as_field((dims.CellDim, dims.KDim), zd_diffcoef)
-    zd_vertoffset = as_field((dims.CellDim, dims.C2E2CDim, dims.KDim), zd_vertoffset)
-    zd_intcoef = as_field((dims.CellDim, dims.C2E2CDim, dims.KDim), zd_intcoef)
-    w = as_field((dims.CellDim, dims.KDim), w)
-    vn = as_field((dims.EdgeDim, dims.KDim), vn)
-    exner = as_field((dims.CellDim, dims.KDim), exner)
-    theta_v = as_field((dims.CellDim, dims.KDim), theta_v)
-    rho = as_field((dims.CellDim, dims.KDim), rho)
-    dual_normal_cell_x = as_field((dims.EdgeDim, dims.E2CDim), dual_normal_cell_x)
-    dual_normal_cell_y = as_field((dims.EdgeDim, dims.E2CDim), dual_normal_cell_y)
-    dual_normal_vert_x = as_field((dims.EdgeDim, dims.E2C2VDim), dual_normal_vert_x)
-    dual_normal_vert_y = as_field((dims.EdgeDim, dims.E2C2VDim), dual_normal_vert_y)
-    primal_normal_cell_x = as_field((dims.EdgeDim, dims.E2CDim), primal_normal_cell_x)
-    primal_normal_cell_y = as_field((dims.EdgeDim, dims.E2CDim), primal_normal_cell_y)
-    primal_normal_vert_x = as_field((dims.EdgeDim, dims.E2C2VDim), primal_normal_vert_x)
-    primal_normal_vert_y = as_field((dims.EdgeDim, dims.E2C2VDim), primal_normal_vert_y)
-    tangent_orientation = as_field((dims.EdgeDim,), tangent_orientation)
-    inverse_primal_edge_lengths = as_field((dims.EdgeDim,), inverse_primal_edge_lengths)
-    inv_dual_edge_length = as_field((dims.EdgeDim,), inv_dual_edge_length)
-    inv_vert_vert_length = as_field((dims.EdgeDim,), inv_vert_vert_length)
-    edge_areas = as_field((dims.EdgeDim,), edge_areas)
-    f_e = as_field((dims.EdgeDim,), f_e)
-    cell_areas = as_field((dims.CellDim,), cell_areas)
+    nudge_max_coeff = (
+        0.075 * DEFAULT_PHYSICS_DYNAMICS_TIMESTEP_RATIO
+    )  # this is done in ICON, so we replicate it here
+    itype_sher = TurbulenceShearForcingType.VERTICAL_HORIZONTAL_OF_HORIZONTAL_VERTICAL_WIND
+    nflat_gradp = 27
```

Comment:
You can set `nflat_gradp` from the `grid_savepoint`

Reply:
Thanks did that now